src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -674,6 +674,31 @@ internal RankingTrainers(RankingCatalog catalog)
             var eval = new RankingEvaluator(Environment, options ?? new RankingEvaluatorOptions() { });
             return eval.Evaluate(data, labelColumnName, rowGroupColumnName, scoreColumnName);
         }
+
+        /// <summary>
+        /// Run cross-validation over <paramref name="numberOfFolds"/> folds of <paramref name="data"/>, by fitting <paramref name="estimator"/>,
```

Comment:
_(appended to unrelated code, to allow for threaded discussion)_

Can you please also add a note to the `samplingKeyColumn` parameter on `TrainTestSplit` and `CrossValidationSplit` stating that if the user is going to use the splits for a Ranking task, then samplingKeyColumn _must_ be the same as the column used for GroupId?

https://github.com/dotnet/machinelearning/blob/b7d0063f581228b2a5e3fae5283ae3618aae83c9/src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs#L393-L466

Issue #2685 also mentions this, and it seems they were thinking about somehow changing the TrainTest API to enforce this for ranking, but they recognized they wouldn't be able to change this once ML.NET v 1.0 came out because it would be a breaking API change. So now, I think the best we can simply do is add a note on the docs :wink: thanks